tests/PHPStan/Analyser/data/call-user-func-array.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace CallUserFuncArray;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @template T
+ * @param T $t
+ * @return T
+ */
+function generic($t) {
+	return $t;
+}
+
+function fun(): int
+{
+	return 3;
+}
+
+function fun3($i, $x, $y): int
+{
+	return 3;
+}
+
+class c {
+	static function m(): string
+	{
+		return 'hello';
+	}
+}
+
+class Foo {
+	/**
+	 * see https://3v4l.org/aOYWW
+	 */
+	function proxy() {
+		$params = [
+			'CallUserFuncArray\generic',
+			[123]
+		];
+
+		assertType('int', call_user_func_array(...$params));
+
+		$params = [
+			'CallUserFuncArray\generic',
+			123,
+			456
+		];
+
+		// cufa expect max 2 args
+		assertType('*NEVER*', call_user_func_array(...$params));
+		assertType('int', call_user_func(...$params));
+	}
```

Comment:
I have no idea yet on how to implement this cases, where only a unpack-array is passed, as I don't know how to split/what to pass into the `FuncCall` - which expects the name and args separated.

all other cases seem to pass right now

Reply:
I realized that in this variadic case I will let the extension just return the `$defaultReturn`